src/overview/actions.js

```diff
@@ -257,12 +269,70 @@ export const showTags = index => (dispatch, getState) => {
 }
 
 export const filterTag = tag => (dispatch, getState) => {
-    const query = selectors.query(getState())
+    const state = getState()
+    const query = selectors.query(state)
+    const filterTags = selectors.filterTags(state)
+
     const transformedTag = `#${tag.split(' ').join('+')} `
 
-    const newQuery = query.includes(transformedTag)
+    const newQuery = query.includes(transformedTag.slice(0, -1))
         ? query.replace(transformedTag, '') // Either remove it, if already there
         : transformedTag + query // or prepend it, if not there
 
-    dispatch(setQuery(newQuery))
+    dispatch(setQueryTagsDomains(newQuery))
+}
+
+export const setQueryTagsDomains = (input, isEnter) => (dispatch, getState) => {
+    const state = getState()
+    const filterTags = selectors.filterTags(state)
+    const filterDomains = selectors.filterDomains(state)
+    let filterStatus = selectors.showFilter(state)
+    const query = input
+
+    const terms = input.toLowerCase().match(/\S+/g) || []
+
+    terms.forEach(term => {
+        if (input[input.length - 1] === ' ' || isEnter) {
```

Comment:
This condition can be placed outside of the `terms.forEach` call, which is going to improve the input performance by a lot. Right now every time you type a character, it will take the input state, lowercase it, split it up into words, then run this function for each word - this can all be skipped to only ever happen when user types space or hits enter